libs/server/Resp/Bitmap/BitmapCommands.cs

```diff
@@ -317,65 +275,56 @@ private bool NetworkStringBitPosition<TGarnetApi>(ref TGarnetApi storageApi)
                 return AbortWithWrongNumberOfArguments(nameof(RespCommand.BITPOS));
             }
 
-            //<[Get Key]>
+            // <[Get Key]>
             var sbKey = parseState.GetArgSliceByRef(0).SpanByte;
 
-            var bSetValSlice = parseState.GetArgSliceByRef(1);
-            Debug.Assert(bSetValSlice.length == 1);
-            var bSetVal = (byte)(bSetValSlice.ReadOnlySpan[0] - '0');
-            Debug.Assert(bSetVal == 0 || bSetVal == 1);
-
-            //Process offsets here if they exist
-            var startOffset = 0; // default is at the start of bitmap array
-            var endOffset = -1; // default is at the end of the bitmap array (negative values indicate offset starting from end)
-            byte bitOffsetType = 0x0; // treat offsets as byte or bit offsets
+            // Validate value
+            var bSetValSlice = parseState.GetArgSliceByRef(1).ReadOnlySpan;
+            if (bSetValSlice.Length != 1 || (bSetValSlice[0] != '0' && bSetValSlice[0] != '1'))
+            {
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_BIT_IS_NOT_INTEGER, ref dcurr,
+                           dend))
+                    SendAndReset();
+                return true;
+            }
 
-            if (parseState.Count > 2)//Start offset exists
+            // Validate start & end offsets, if exist
+            if (parseState.Count > 2)
             {
-                if (!parseState.TryGetInt(2, out startOffset) || (parseState.Count > 3 && !parseState.TryGetInt(3, out endOffset)))
+                if (!parseState.TryGetInt(2, out _) ||
+                    (parseState.Count > 3 && !parseState.TryGetInt(3, out _)))
                 {
-                    while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref dcurr, dend))
+                    while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_VALUE_IS_NOT_INTEGER, ref dcurr,
+                               dend))
                         SendAndReset();
                     return true;
                 }
             }
 
+            // Validate offset range type (BIT / BYTE), if exists
             if (parseState.Count > 4)
             {
                 var sbOffsetType = parseState.GetArgSliceByRef(4).ReadOnlySpan;
-                bitOffsetType = sbOffsetType.EqualsUpperCaseSpanIgnoringCase("BIT"u8) ? (byte)0x1 : (byte)0x0;
+                if (!sbOffsetType.EqualsUpperCaseSpanIgnoringCase("BIT"u8) &&
```

Comment:
Why can this not be passed as a parameter in input, similar to arg1 etc. and if the backend does not find it (not sure why, maybe because of GarnetAPI code path?), it can reparse it from the parse state.

Reply:
Leaving as-is for now